packages/adblocker-electron-preload/preload.ts

```diff
@@ -10,10 +10,13 @@ import { ipcRenderer } from 'electron';
 
 import { DOMMonitor, IBackgroundCallback, IMessageFromBackground } from '@cliqz/adblocker-content';
 
-function getCosmeticsFilters(data: IBackgroundCallback) {
-  setTimeout(() => {
+function getCosmeticsFiltersFirst(): string[] | null {
+  return ipcRenderer.sendSync('get-cosmetic-filters-first', window.location.href);
+}
+function getCosmeticsFiltersUpdate(data: Omit<IBackgroundCallback, 'lifecycle'>) {
+  setImmediate(() => {
```

Comment:
This function is no longer present in modern browsers, and as such breaks the latest release on my system.
https://developer.mozilla.org/en-US/docs/Web/API/Window/setImmediate#browser_compatibility
@remusao

Reply:
oh shit I really don't know that. I'll fix it as soon as possible:sob:

Reply:
If I replace it with setTimeout I no longer get errors, but I still see ads on youtube, more than before... So maybe that's not the correct replacement. I had to inject the blocker in a custom way though, so I'll look into that first and get back when I know more.

Reply:
> If I replace it with setTimeout I no longer get errors, but I still see ads on youtube, more than before

Weird... Isn't the difference between `setImmediate` and `setTimeout` just about 4ms?

Reply:
I have a thought, but do we still need this setImmediate?
We separated the method called in the first call, and since this method is called at DOMContentLoaded, I feel like the processing won't change whether we execute it immediately or insert a wait.

@Jelmerro 
By the way, I'm curious about the issue of more ads showing up than before. Could you show me the Console and Network tabs in Devtools if you don't mind?

Reply:
Should be, but I still have to update the header injection, as I already have a listener for onHeadersreceived and onBeforeRequest I have to do some manual work to update it. I just wanted to mention the error I got on setImmediate, which is not related to the main process code changes, sadly I don't know why 4ms would make such a big difference, or if it's even related to seeing the youtube ads.

Reply:
May I know in what environment did you perform the test? Could you also tell me the HTML file that you opened in Electron? (The difference in how YouTube is opened might be the cause.)

Reply:
I do see the usual blocking happening in the console, but I do not remember getting 2 ads before starting a video. 
![image](https://user-images.githubusercontent.com/1696674/229347137-f05345fb-b24e-40fe-95fc-babbbd5d3c57.png)
There do not seem to be any errors here after replacing setImmediate with setTimeout, but I do get 2 video ads before each video.

Reply:
I'm testing this for my project [Vieb](https://github.com/Jelmerro/Vieb/), where I made the following changes compared to current master:

- ran `python update.py` t get the new adblocker update package
- replaced setImmediate with setTimeout in the preload package
- The following code changes to index.js to reflect the changes to the listener:
![image](https://user-images.githubusercontent.com/1696674/229347311-64e6b14a-e776-4d51-9697-1b6ec0890696.png)

Again, I'm fairly certain the issues are due to the custom setup and probably not the adblocker package itself, but so far I have yet to see cosmetic filtering work with the new release, even after fixing the setImmediate issue.

Reply:
https://github.com/ghostery/adblocker/pull/3166
I submitted a PR for the fix. I'll also test it in the Vieb environment, so there might be some parts that still need to be fixed.

Reply:
@Jelmerro 
I briefly looked at the source code, but it seems like the block list is pulled from `/app/blocklists/list.json`, which contains two files: `easylist.txt` and `easyprivacy.txt`. However, the scriptlet to block video ads at the beginning is not included in these two files, so if there are no other lists being read, I don't think it can block the initial video ad. Specifically, the following entries are required:
```
youtube.com, youtubekids.com, youtube-nocookie.com##+js(json-prune, [].playerResponse.adPlacements [].playerResponse.playerAds playerResponse.adPlacements playerResponse.playerAds adPlacements playerAds)
youtube.com, youtubekids.com, youtube-nocookie.com##+js(set, ytInitialPlayerResponse.adPlacements, undefined)
youtube.com, youtubekids.com, youtube-nocookie.com##+js(set, playerResponse.adPlacements, undefined)
```

Reply:
Thank you for looking into this, I will see if this makes it work. Do you happen to know which list used by ghostery includes these by default?

Reply:
https://github.com/ghostery/adblocker/blob/master/packages/adblocker/src/fetch.ts
The URLs that are eventually loaded by `fromPrebuiltAdsAndTracking` and other functions are hard-coded here.

Reply:
Found, it it's this one: `packages/adblocker/assets/ublock-origin/filters.txt`:
![image](https://user-images.githubusercontent.com/1696674/229349603-117e398e-53d6-4211-8a09-ad4eddcea3ed.png)